### PR TITLE
Component Reusablity

### PR DESCRIPTION
**Pull request by Mohitktater**
Hello 
please check the code for select box reuse functionality

### CODE REVIEW

File: src/components/select/Search.js
Comment (Delwalt) [inline src/components/select/Search.js:1]: why do we have a this file ? All logic of this file can inside the index file.

**Review by Delwalt — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,114 @@
-import Main from "./components/select/";
 import "./App.css";
+import React, { useState } from "react";
+import Select from "./components/select/";
 
 function App() {
+  const softwareOptions = [
```
[the remark this answers is not in the document]
Comment (Mohitktater) [reply src/App.js:6]: Resolved

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,114 @@
...
+    { label: "Hyundai i20", value: "2" },
+    { label: "Tata Nexon", value: "3" },
+    { label: "Mahindra Thar", value: "4" },
+    { label: "Toyota Fortuner", value: "5" },
+    { label: "Honda City", value: "6" },
+    { label: "Kia Seltos", value: "7" },
+    { label: "Renault Kiger", value: "8" },
+    { label: "MG Hector", value: "9" },
+  ];
+  const TeamOptions = [
```
[the remark this answers is not in the document]
Comment (Mohitktater) [reply src/App.js:28]: Resolved

File: src/components/select/Search.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
-import React, { useState } from "react";
+import React, { useState, useRef, useEffect } from "react";
+
 import "./search.css";
 
 export const SearchableSelect = ({ options, onSelect }) => {
```
[the remark this answers is not in the document]
Comment (Mohitktater) [reply src/components/select/Search.js:5]: Resolved

File: src/components/select/Search.js
Comment (Mohitktater) [reply src/components/select/Search.js:1]: Resolved

File: src/components/select/Search.js [hunk 4]
```diff
@@ -21,18 +23,31 @@ export const SearchableSelect = ({ options, onSelect }) => {
 
   const handleOptionClick = (option) => {
     setSelectedOption(option);
-    onSelect(option.value);
+    onSelect(option);
     setFilteredOptions([]);
     handleShowList();
   };
 
   const handleShowList = () => {
```
[the remark this answers is not in the document]
Comment (Mohitktater) [reply src/components/select/Search.js:31]: Resolved

File: src/App.js [hunk 5]
```diff
@@ -1,10 +1,114 @@
...
+    { label: "Nikhil Joshi", value: "7" },
+    { label: "Sonia Shah", value: "8" },
+    { label: "Rajesh Koothrappali", value: "9" },
+  ];
+
+  const [selectedOptionSoftware, setSelectedOptionSoftware] = useState();
+  const [selectedOptionCar, setSelectedOptionCar] = useState();
+  const [selectedOptionTeam, setSelectedOptionTeam] = useState();
+
+  const handleSelectChangeSoftware = (Option) => {
```
[the remark this answers is not in the document]
Comment (Mohitktater) [reply src/App.js:44]: Resolved

**Pull request closed without merging**